bindings/python/src/ser.rs

```diff
@@ -74,6 +74,23 @@ fn get_object_type_from_object(object: *mut pyo3::ffi::PyObject) -> ObjectType {
     }
 }
 
+fn get_type_name(object_type: *mut pyo3::ffi::PyTypeObject) -> String {
+    let type_name = unsafe { CStr::from_ptr((*object_type).tp_name).to_string_lossy() };
```

Comment:
`type_name` is `Cow<str>`, and likely its borrowed variant (as the type name is likely to be a valid UTF-8 string). Allocating a new `String` on the following line is not needed when it is passed to `format!` inside `check_type_is_str` later on.

This code path is quite rare, and it is just saves a single allocation, so feel free to skip this.

Reply:
In the case of `Cow` it is necessary to specify a `'static` lifetime parameter, otherwise the compiler gets a error. 
Which does not seem very true, since the type can be deleted.

I'm not very good at rust yet, maybe specifying it as static would be correct.

Reply:
Indeed, the lifetime is bound to the Python interpreter memory flow, which we can assume to be `'static` at this point (which is not 100% true, as you mentioned). CPython guarantees that GIL is held when this extension gets the control - no other thread can remove the type from Python's heap. Here we also don't spin anything else that can affect the type, so, given these circumstances I'd say it should be safe to use `Cow<'static, str>` in this case. The underlying slice will be accessed only a little longer (`format!` will create an owned string just after this). Technically, we already hold `Cow<'static, str>` here in `type_name`, such a change won't change the implementation conceptually (it is still "type pointer" -> "CStr" -> "Cow" -> "String")

Reply:
Or, better - it should be the `'p` lifetime, which we got from the original `&PyAny` "owned reference", that is GIL-bound.

Reply:
Okay, thanks for such a detailed answer.
In that case, yes, it looks like we won't break any invariants, and the reference to the type name will only be stored until the `format!` execution.

I can fix that in a new pull request.

Reply:
Didn't see that you had already fixed it)
Thank you!

Reply:
Sorry, I forgot to let you know about it :)

Reply:
I'll make a new release soon